http/oidc/src/test/java/org/wildfly/security/http/oidc/KeycloakConfiguration.java

```diff
@@ -157,12 +188,60 @@ private static ClientRepresentation createWebAppClient(String clientId, String c
         client.setRedirectUris(Arrays.asList("http://" + clientHostName + ":" + clientPort + "/" + clientApp));
         client.setEnabled(true);
         client.setDirectAccessGrantsEnabled(directAccessGrantEnabled);
+
         if (allowedOrigin != null) {
             client.setWebOrigins(Collections.singletonList(allowedOrigin));
         }
+        OIDCAdvancedConfigWrapper oidcAdvancedConfigWrapper = OIDCAdvancedConfigWrapper.fromClientRepresentation(client);
+        oidcAdvancedConfigWrapper.setUseJwksUrl(false);
+        KEYSTORE_CLASSPATH = Objects.requireNonNull(KeycloakConfiguration.class.getClassLoader().getResource("")).getPath();
+        String rsaCert = generateKeyStoreFileAndGetCertificate("Rsa", 2048, KEYSTORE_CLASSPATH + RSA_KEYSTORE_FILE_NAME, "SHA256WITHRSA");
+        client.getAttributes().put("jwt.credential.certificate", rsaCert);
         return client;
     }
 
+    static String generateKeyStoreFileAndGetCertificate(String algorithm, int keySize, String keystorePath, String certSignAlg) throws GeneralSecurityException, IOException, OperatorCreationException {
+        KeyPairGenerator keyGen = KeyPairGenerator.getInstance(algorithm);
+        keyGen.initialize(keySize);
+        KeyPair keys = keyGen.generateKeyPair();
+        X509Certificate cert = createCertificate(keys, certSignAlg);
+        KeyStore keystore = createKeyStore(cert, keys.getPrivate());
+        try (FileOutputStream fileOutputStream = new FileOutputStream(keystorePath)) {
+            keystore.store(fileOutputStream, KEYSTORE_PASS.toCharArray());
+        }
+
+        KeyStoreConfig keyStoreConfig = new KeyStoreConfig();
+        keyStoreConfig.setKeyAlias(KEYSTORE_ALIAS);
+        keyStoreConfig.setStorePassword(KEYSTORE_PASS);
+        keyStoreConfig.setKeyPassword(KEYSTORE_PASS);
+        keyStoreConfig.setRealmCertificate(true);
+        return Base64.getEncoder().encodeToString(cert.getEncoded());
+    }
+
+    private static KeyStore createKeyStore(X509Certificate certificate, PrivateKey privateKey) throws IOException, GeneralSecurityException {
+        KeyStore keyStore = createEmptyKeyStore();
+        keyStore.setCertificateEntry("jwtKeystore", certificate);
+        keyStore.setKeyEntry("jwtKeystore", privateKey, "password".toCharArray(), new Certificate[]{certificate});
+        return keyStore;
+    }
+
+    private static KeyStore createEmptyKeyStore() throws IOException, GeneralSecurityException {
+        KeyStore keyStore = KeyStore.getInstance("PKCS12");
+        keyStore.load(null,null);
+        return keyStore;
+    }
+
+    private static X509Certificate createCertificate(KeyPair keyPair, String certSignAlg) throws GeneralSecurityException, OperatorCreationException {
```

Comment:
I think we have a CAGenerationTool test class that can be used to generate certs needed for tests. Can we make use of that here?